File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    if (s.length === 0) return '';
    let longest = 1;
    let subStr = s[0];
    for (let i = 1; i < s.length; i++) {
        if (subStr.includes(s[i])) {
            const index = subStr.indexOf(s[i]);
            subStr = subStr.slice(index + 1) + s[i];
        } else {
            subStr += s[i];
        }
        longest = Math.max(longest, subStr.length);
    }
    return longest;
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
      if len(s) == 0:
        return 0
      
      longest = 1
      stack = [ s[0] ]
      i = 1
      
      while i < len(s):
        if s[i] in stack:
          #Find index of repeated character
          idx = stack.index(s[i])
          stack = stack[idx+1:]
          stack.append(s[i])
        else:
          stack.append(s[i])
          
        longest = max(longest, len(stack))
        i += 1
        
      return longest